lib/interfaces.js

```diff
@@ -110,12 +115,16 @@ function httpsFactory(iface, fallback, mosca) {
     throw new Error("missing credentials for https server");
   }
 
-  if (credentials.keyPath) {
-    credentials.key = fs.readFileSync(credentials.keyPath);
+  if (credentials.key) {
+    if (fs.existsSync(credentials.key)) {
+      credentials.key = fs.readFileSync(credentials.key);
+    }
```

Comment:
I was probably not clear.
The options `keyPath`, `certPath` and `caPath` are the only ones that needs to call `readFileSync`.
The options `key`, `cert` and  `ca`, should not.